pkg/registry/thirdpartyresourcedata/codec.go

```diff
@@ -328,6 +427,8 @@ func (t *thirdPartyResourceDataCodec) EncodeToStream(obj runtime.Object, stream
 		return nil
 	case *unversioned.Status:
 		return t.delegate.EncodeToStream(obj, stream, overrides...)
+	case *unversioned.APIResourceList:
```

Comment:
It seems a little odd to whitelist delegation.  I'm not terribly familiar with the flow, but I'd've thought this whole thing would be a pass-through in cases where it wasn't dealing with exactly what it wanted.